src/randomgroups/algorithm.py

```diff
@@ -77,20 +79,29 @@ def _compute_assortativity_score(
 
     Args:
         matching (list): Matching.
-        faculty_multiplier (float): Multiplier determining how much faculty members
-            want to stay in the same group.
+         mixing_multiplier (float, List): Multiplier determining how many members of
+            different status want to stay in the same group. Positive values favor
+            assortative matchings, negative values favor mixed matchings.
 
     Returns:
         float: The corresponding score.
 
     """
+    # check if group has more than one status
+    statuses_columns = [col for col in matching[0].columns if "status" in col]
+
     score = 0.0
     for group in matching:
-        if len(group.status.unique()) > 1:
-            wants_assortative = 1 - group.wants_mixing
-            wants_assortative.loc[group.status == "faculty"] *= faculty_multiplier
-            score += wants_assortative.mean()
-
+        if "wants_mixing" not in group.columns:
+            # if not, assume that no-one wants mixing
+            group["wants_mixing"] = 0
```

Comment:
You're doing this for every group, but this should be the same for all groups, right? In this case, I'd prefer that this is handled outside this function. Ideally, this would be an argument of `create_matching`, but that you don't need to do.

Reply:
I think as an argument `wants_mixing` is a little bit redundant since if you want everybody to mix you can just set `assortative_matching=False` and if only some want mixing it should be specified in the names-file.